x-pack/agentbeat/capabilities_linux.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"kernel.org/pub/linux/libs/security/libcap/cap"
+
+	"github.com/elastic/beats/v7/x-pack/libbeat/common/utils"
+)
+
+type capProc interface {
+	GetFlag(vec cap.Flag, val cap.Value) (bool, error)
+	SetFlag(vec cap.Flag, enable bool, val ...cap.Value) error
+	SetProc() error
+}
+
+var (
+	// for unit-testing
+	capProcFunc = func() capProc {
+		return cap.GetProc()
+	}
+)
+
+func initCapabilities() {
+	isRoot, err := utils.HasRoot()
```

Comment:
This entire block of code is Linux specific, you don't need to also duplicate the Windows and MacOS `HasRoot` implementations. You can just make the implementation of `initCapabilities` a no-op on those OSes instead.

If we want to share `HasRoot` put them in elastic-agent-libs, the implementation particularly for Windows is non-trivial and has had bugs recently.

Reply:
I've removed it for other OSes.
Can you take a look?